core/client/src/main/java/alluxio/client/block/BlockMasterClient.java

```diff
@@ -18,17 +18,38 @@
 import alluxio.wire.WorkerInfo;
 
 import java.io.IOException;
+import java.net.InetSocketAddress;
 import java.util.List;
 
 import javax.annotation.concurrent.ThreadSafe;
+import javax.security.auth.Subject;
 
 /**
- * A client to use for interacting with a Block Master.
+ * A client to use for interacting with a block master.
  */
 @ThreadSafe
 public interface BlockMasterClient extends Client {
 
   /**
+   * Factory for {@link BlockMasterClient}.
+   */
+  class Factory {
+
+    private Factory() {} // prevent instantiation
+
+    /**
+     * Factory method for {@link BlockMasterClient}.
+     *
+     * @param subject the parent subject, set to null if not present
```

Comment:
Would it make sense to make another `create` method which does not take a subject instead of asking the caller to explicitly pass in null?